### PR TITLE
Add rule ParameterTypeHint

### PR DESCRIPTION
**Pull request by rdss-sknott**
[SlevomatCodingStandard.TypeHints.ParameterTypeHint](https://github.com/slevomat/coding-standard#slevomatcodingstandardtypehintsparametertypehint-)

### CODE REVIEW

File: src/main/php/Github/Commands/PullCommentRefreshCommand.php [hunk 1]
```diff
@@ -160,9 +154,9 @@ private function createComment(array $arguments): void
     /**
      * Updates an comment which is not stale by the given arguments from commandline.
      *
-     * @param array $ownComments
-     * @param array $staleComments
-     * @param array $arguments
+     * @param array<mixed> $ownComments
```
Comment (rdss-pwacht) [inline src/main/php/Github/Commands/PullCommentRefreshCommand.php:157]: Why do you use here mixed?

**Review by rdss-pwacht — APPROVED** [review]
This change is welcome from my perspective 😉

File: src/main/php/Github/Commands/PullCommentRefreshCommand.php (hunk 1 above)
Comment (rdss-skirejewski) [reply src/main/php/Github/Commands/PullCommentRefreshCommand.php:157]: This array can be anything.
I guess array<mixed> is the only choice if you want to satisfy also higher levels 6.
But it has to be `array<mixed`, `mixed` will throw an error in level 6.
If being compatible with higher levels is desired in this major version, 
I would suggest to also transform the `array<string|bool|int|float|array|null>` into `array<mixed>`
because by putting an `array` as a subtype, this will force error in level 6.

**Review by rdss-skirejewski — APPROVED** [review]

File: src/main/php/Github/Commands/PullCommentRefreshCommand.php (hunk 1 above)
Comment (rdss-skirejewski) [reply src/main/php/Github/Commands/PullCommentRefreshCommand.php:157]: argh somehow i deleted my comment ...
tldr;
`array<mixed>` should be the preferred approach here as this array can contain everything.
If conformity with higher phpstan levels is desired in here. I would suggest to transform `array<string|bool|int|float|array|null> $arguments` into `array<mixed>` because the subtype of array will throw an exception in PHPstan level 6 and above.

Comment (rdss-sknott) [reply src/main/php/Github/Commands/PullCommentRefreshCommand.php:157]: My reasoning in this case is, that this array is part of a json response from the github api and I have no control over the payload.

Comment (rdss-pwacht) [reply src/main/php/Github/Commands/PullCommentRefreshCommand.php:157]: then is better how @rdss-skirejewski suggest, because mixed also contain callable and this isn't here

**Review by jhofm — APPROVED** [review]

**Review by rdss-dkontsevoi — APPROVED** [review]
